frontend/src/features/calender/CalenderReducer.ts

```diff
@@ -0,0 +1,74 @@
+import { getWeekRange } from "./Calender.util";
+
+interface CalenderState {
+  calendarDate: Date; // 렌더링되는 기준 날짜
+  selectedDate: Date | null; // 현재 선택된 날짜
+  selectedWeek: Date[]; // 현재 선택된 날짜가 포함된 주
+}
+
+type CalenderAction =
+  | { type: "SELECT_DATE"; payload: Date }
+  | { type: "CHANGE_MONTH"; payload: Date; direction: "next" | "prev" }
+  | { type: "SET_SELECTED_WEEK"; payload: Date };
+
+const initialState: CalenderState = {
+  selectedDate: new Date(),
+  calendarDate: new Date(),
+  selectedWeek: getWeekRange(new Date()),
+};
+
+const calenderReducer = (
+  state: CalenderState,
+  action: CalenderAction,
+): CalenderState => {
+  switch (action.type) {
+    case "SELECT_DATE": {
+      const clickedDate = action.payload;
+      const newState = { ...state, selectedDate: clickedDate };
+
+      // 주 설정
+      newState.selectedWeek = getWeekRange(clickedDate);
+
+      // 다른 월이면 월 변경
+      if (
+        clickedDate.getMonth() !== state.calendarDate.getMonth() ||
+        clickedDate.getFullYear() !== state.calendarDate.getFullYear()
+      ) {
+        newState.calendarDate = new Date(
+          clickedDate.getFullYear(),
+          clickedDate.getMonth(),
+          1,
+        );
+      }
+
+      return newState;
+    }
```

Comment:
newState에 selectedDate를 설정한 후에, selectedWeek를 또 다시 분리하여 설정하시는 이유는 무엇인가요?
주 설정을 모두 진행하고 -> 그걸 기반으로 newState에 한 번에 업데이트하는 방식도 괜찮아보입니다!

```suggestion
const newState = { 
...state, 
selectedDate: clickedDate,
selectedWeek: getWeekRange(clickedDate)
 };
```

Reply:
기존 로직에서 월 변경 로직 추가하다가 놓친 것 같네요 감사합니다 !